### PR TITLE
Integrate GameMaster with services. Part I

### PR DESCRIPTION
**Pull request by R1ZZU**

### CODE REVIEW

File: src/game-master/components/GameMasterLayout.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { h } from 'preact';
+import { withGroupedChildren } from '../../components/with-grouped-children-hoc';
+
+const PureGameMasterLayout = ({ components }) => console.log(components) || (
```
Comment (DaQuirm) [inline src/game-master/components/GameMasterLayout.js]: Is this `console.log` intentional?

File: src/game-master/containers/RoundPhaseButtonContainer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    roundStarted,
+}), {
+    triggerRoundStart,
+    triggerRoundEnd,
+})(({ roundStarted, triggerRoundStart, triggerRoundEnd }) => (
+    <Button
+        iconLeft={roundStarted ? 'stop' : null }
+        iconRight={!roundStarted ? 'arrow-right' : null }
+        text={roundStarted ? 'STOP LEVEL' : 'NEXT LEVEL'}
+        onClick={roundStarted ? () => triggerRoundEnd() : () => triggerRoundStart() }
```
Comment (Shastel) [inline src/game-master/containers/RoundPhaseButtonContainer.js]: Is wrapping of AC in additional function necessary?

File: src/shared/middlewares/phoenix-receiver-middleware.js [hunk 3]
```diff
@@ -60,12 +62,14 @@ export const phoenixReceiverMiddleware = store => next => action => {
...
+                timeLimit: message.puzzleOptions.timeLimit,
                 expectedSelection: JSON.parse(message.expected),
             });
             break;
         }
 
-        case MESSAGE_NAME.playerSessionState: {
+        case MESSAGE_NAME.playerSessionState:
+        case MESSAGE_NAME.gameMasterSessionState: {
             spreadSessionState(store, message);
```
Comment (DaQuirm) [inline src/shared/middlewares/phoenix-receiver-middleware.js]: image

File: src/shared/middlewares/phoenix-receiver-middleware.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ export const phoenixReceiverMiddleware = store => next => action => {
     }
 
     const { message } = parseMessage(action.payload.data);
+    console.log(message);
```
Comment (DaQuirm) [inline src/shared/middlewares/phoenix-receiver-middleware.js]: Another console.log, just asking 😺

File: src/game/middlewares/phoenix-solution-middleware.js [hunk 5]
```diff
@@ -29,6 +29,11 @@ export const phoenixSolutionMiddleware = state => next => action => {
         store.dispatch(selectionAction);
     }
 
+    if (message.name === MESSAGE_NAME.puzzleChanged) {
+        const clearSelectorAction = clearSelector();
```
Comment (DaQuirm) [inline src/game/middlewares/phoenix-solution-middleware.js]: Maybe call this one directly?

File: src/game-master/containers/RoundPhaseButtonContainer.js (hunk 2 above)
Comment (R1ZZU) [reply src/game-master/containers/RoundPhaseButtonContainer.js]: without this wrapper payload of created action is `MouseEvent` 👎

Comment (Shastel) [reply src/game-master/containers/RoundPhaseButtonContainer.js]: :(

File: src/game/middlewares/phoenix-solution-middleware.js (hunk 5 above)
Comment (R1ZZU) [reply src/game/middlewares/phoenix-solution-middleware.js]: I prefer having separate var for easier debug. Also it looks more consistent with previous case, where one-line call will look not so pretty

Comment (DaQuirm) [reply src/game/middlewares/phoenix-solution-middleware.js]: 👍

**Pull request merged**